### DISCUSSION TITLE
How do I change color of singleValue when hovering on control?

### DISCUSSION THREAD

**Discussion by cpnvr** [JedWatson/react-select#4691] [Q&A] [answered]
Is it possible to change the color of `singleValue` text when hovering anywhere over the control div?

Right now if I add a hover state to `singleValue` itself, it only takes place if I'm mousing over the text itself -- not the entire Select component, which is my goal.

I've tried everything myself and searched SO as well, so I'm going here as a last resort and would very much appreciate your help 🙏

**Comment by Rall3n** [comment, marked as answer]
Hello @cpnvr,

I would suggest a combination of `classNamePrefix` and the styling api.

You use the `classNamePrefix` prop to apply fixed classes to the inner components (`SingleValue` would have `PREFIX__single-value`). With this you can apply a `:hover` style onto the `Container` component to target the `SingleValue` by class.

```jsx
<Select
    classNamePrefix='select'
    styles={{
        container: (provided) => ({
            ...provided,
            '&:hover .select__single-value': {
                color: 'red'
            }
        })
    }}
/>
```

**Reply by cpnvr** [reply]
I legit would never have figured that out on my own. Works perfectly \^_^/

The one thing that I added to complete the feature is a similar rule for `placeholder`:

```
container: (provided, state) => ({
      // wrapper for control
      ...provided,
      "&:hover .select__single-value, &:hover .select__placeholder": {
        color: "purple",
      },
    }),
```

Thank you so very much, @Rall3n

**Reply by Rall3n** [reply]
@cpnvr Please don't forget to mark this as the answer, so everyone can see that your question is resolved.

**Reply by cpnvr** [reply]
@Rall3n Happy to.... but I noticed that there's an edge case.

The color reverts to its un-hovered style when the mouse drifts into the gap between `control` and the menu below it. The code I've written to try and address this isn't taking effect:

```
// y u no work

container: (provided, state) => ({
      // wrapper for control
      ...provided,
      "&:hover .select__single-value, &:hover .select__placeholder": {
        color: colors.darkBlue,
      },
      ".select__control": {
        borderColor: state.isFocused && colors.orange,
        color: state.isFocused && colors.orange,
      },
      ".select__single-value": {
        color: state.isSelected && colors.orange,
      },
    }),
...
    singleValue: (provided, state) => ({
      // active text label input selector
      ...provided,
      color: colors.blue,
      textTransform: "uppercase",
      fontSize: ".875rem",
    }),
```

### Screenshots:

Default State:

<img width="867" alt="Screen Shot 2021-07-15 at 11 38 48 AM" src="https://user-images.githubusercontent.com/84046389/125840233-b421a441-6bfb-4dcd-9b22-867e8a880fd2.png">

Hover before click:

<img width="760" alt="Screen Shot 2021-07-15 at 11 39 30 AM" src="https://user-images.githubusercontent.com/84046389/125840291-aba06b81-3719-4e41-a8ea-ea6171691f78.png">

Hover after click:

<img width="851" alt="Screen Shot 2021-07-15 at 11 36 11 AM" src="https://user-images.githubusercontent.com/84046389/125840442-a4d0ed73-5099-4026-aa95-ef9ffc05cb26.png">


Nooooooooooooooooo (menu open but no hover: this is the edge case):

<img width="880" alt="Screen Shot 2021-07-15 at 11 36 15 AM" src="https://user-images.githubusercontent.com/84046389/125840518-92c308ae-3501-474a-9750-ee9bc7604665.png">

Hover on menu -- this works now:

<img width="767" alt="Screen Shot 2021-07-15 at 11 36 18 AM" src="https://user-images.githubusercontent.com/84046389/125840646-84f32988-d827-417b-a776-4c65626973d6.png">


Any suggestions?

**Reply by Rall3n** [reply]
@cpnvr That is how `position: absolute` works. The dropdown is still part of the container, but does not affect its bounds.

You could reduce the gap between the dropdown and the container if you want to by removing `margin-top` from the `menu` styles.
```jsx
styles={{
  menu: ({ marginTop, ...provided }) => provided
}}
```

**Comment by cpnvr** [comment]
Okay I figured it out -- it was `isFocused` not `isSelected` that did the trick:

```
control: (provided, state) => ({
      ...
      "&:hover": {
        borderColor: colors.orange,
        color: state.isFocused ? colors.orange : colors.blue,
      },
      ".select__single-value": {
        color: state.isFocused && colors.orange,  // awwwww yeahhhhh
      },
    }),
```
